File: References/Discoverify/backend/models/userSchema.js
const mongoose = require('mongoose');

const { Schema } = mongoose;

const userSchema = new Schema({
  userId: { type: String, required: true, unique: true },
  refreshToken: { type: String, required: true, unique: true },
  playlistId: { type: String },
  lastUpdated: { type: Date },
  playlistOptions: {
    seeds: { type: [String], default: ['ST', 'ST', 'MT', 'MT', 'MT'] },
    acousticness: { type: [Number, Number], default: [10, 90] },
    danceability: { type: [Number, Number], default: [10, 90] },
    energy: { type: [Number, Number], default: [10, 90] },
    instrumentalness: { type: [Number, Number], default: [10, 90] },
    popularity: { type: [Number, Number], default: [50, 100] },
    valence: { type: [Number, Number], default: [10, 90] },
  },
  stripeId: { type: String, required: false },
  grandmothered: { type: Boolean, default: false },
});

const userModel = mongoose.model('users', userSchema, 'users');

module.exports = userModel;

File: References/Discoverify/backend/models/userSchema.py
# from mongoengine import Document, StringField, ListField, FloatField, IntField, BooleanField

# class User(Document):
#     user_id = StringField(required=True, unique=True)
#     refresh_token = StringField(required=True, unique=True)
#     playlist_id = StringField()
#     last_updated = DateTimeField()
#     playlist_options = {
#         'seeds': ListField(StringField(), default=['ST', 'ST', 'MT', 'MT', 'MT']),
#         'acousticness': ListField(FloatField(), default=[10, 90]),
#         'danceability': ListField(FloatField(), default=[10, 90]),
#         'energy': ListField(FloatField(), default=[10, 90]),
#         'instrumentalness': ListField(FloatField(), default=[10, 90]),
#         'popularity': ListField(IntField(), default=[50, 100]),
#         'valence': ListField(FloatField(), default=[10, 90]),
#     }
#     stripe_id = StringField()
#     grandmothered = BooleanField(default=False)

#     meta = {'collection': 'users'}